### PR TITLE
Storkctl subcommands for applications crds

### PR DESCRIPTION
**Pull request by disrani-px**
**What type of PR is this?**
Enhancement

**What this PR does / why we need it**:

**Does this PR change a user-facing CRD or CLI?**:
Adds new subcommands to `storkctl`

**Is a release note needed?**:
Include with the feature release notes

**Does this change need to be cherry-picked to a release branch?**:
No

### CODE REVIEW

File: pkg/storkctl/applicationclone_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	name := "dummy-name"
+	cmdArgs := []string{"create", "clones", "-n", srcNamespace, "--sourceNamespace", srcNamespace,
+		"--destinationNamespace", dstNamespace, name, "-w"}
+
+	expected := "ApplicationClone dummy-name started successfully\n" +
+		"STAGE\t\tSTATUS              \n" +
+		"\t\t                    \n" +
+		"Volumes\t\tSuccessful          \n" +
+		"ApplicationClone dummy-name completed successfully\n"
+	go setApplicationCloneStatus(name, srcNamespace, false, t)
```
Comment (ram-infrac) [inline pkg/storkctl/applicationclone_test.go]: I'm planning to customise retryTimeout and retryInterval for unit test purpose 
eg. `storkctl create migrations -w 2,10 (2 seconds retry, 10 Min wait)

File: pkg/storkctl/backuplocation.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+			s3BackupLocations := &storkv1.BackupLocationList{}
+			azureBackupLocations := &storkv1.BackupLocationList{}
+			googleBackupLocations := &storkv1.BackupLocationList{}
+			unknownBackupLocations := &storkv1.BackupLocationList{}
+			for _, bl := range backupLocations.Items {
+				switch bl.Location.Type {
+				case storkv1.BackupLocationS3:
+					if !showSecrets && bl.Location.S3Config != nil {
+						bl.Location.S3Config.SecretAccessKey = hiddenString
```
Comment (ram-infrac) [inline pkg/storkctl/backuplocation.go]: do we also need to <hidde>  access key -id as well
Same for other cloud provider Storage-Account-Name and Project-ID

File: pkg/storkctl/applicationrestore.go [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+		Use:     applicationRestoreSubcommand,
+		Aliases: applicationRestoreAliases,
+		Short:   "Start an applicationRestore",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				util.CheckErr(fmt.Errorf("exactly one name needs to be provided for applicationrestore name"))
+				return
+			}
+			if backupLocation == "" {
+				util.CheckErr(fmt.Errorf("need to provide BackupLocation to use for backup"))
```
Comment (ram-infrac) [inline pkg/storkctl/applicationrestore.go]: for restore*

File: pkg/storkctl/applicationclone_test.go (hunk 1 above)
Comment (disrani-px) [reply pkg/storkctl/applicationclone_test.go]: Unit tests can change private variables so we can just reduce the variables for the poll times. I already reduced it for the migration UT

File: pkg/storkctl/backuplocation.go (hunk 2 above)
Comment (disrani-px) [reply pkg/storkctl/backuplocation.go]: Only trying to hide the secret information here.

**Pull request merged**